.github/workflows/release.yml

```diff
@@ -0,0 +1,44 @@
+name: build and push docker images on release.
+
+on:
+  release:
+    types: [ published ]
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+
+      - name: Build and push
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          platforms: linux/amd64,linux/arm64,linux/arm/v7
+          push: true
+          tags: ghcr.io/fungover/haze/server:latest
```

Comment:
Will this code create an image called server tagged with latest and no other tag names? If so can we add some code that will create also a tag for the docker image based on the git tag name associated with the release that triggered this run?

So if I create a tag called 1.0.0 in git we also want the docker image pushed under the name server:1.0.0

Would also be nice if the name of the docker image would be ghcr.io/fungover/haze:latest on github packages and on docker hub only username/haze:latest

[Check here for some instructions on generating tags in different format.](https://docs.docker.com/build/ci/github-actions/examples/#manage-tags-and-labels)

Reply:
> Would also be nice if the name of the docker image would be ghcr.io/fungover/haze:latest on github packages and on docker hub only username/haze:latest

We have gotten it to work with adding the correspondent tag in git, but cant seem to remove the ghcr.io before fungover/haze:latest on the name of docker image on dockerhub, any suggestions?

Reply:
@kappsegla you requested changes have been fixed. please have another look.

Reply:
Check here: https://docs.docker.com/build/ci/github-actions/examples/#manage-tags-and-labels

Looks like it should exist an action called docker/metadata-action

```
id: meta
        uses: docker/metadata-action@v4
        with:
          # list of Docker images to use as base name for tags
          images: |
            name/app
            ghcr.io/username/app
          # generate Docker tags based on the following events/attributes
          tags: |
            type=schedule
            type=ref,event=branch
            type=ref,event=pr
            type=semver,pattern={{version}}
            type=semver,pattern={{major}}.{{minor}}
            type=semver,pattern={{major}}
            type=sha
```
And according to the example it looks like it should be able to handle both docker hub and github automatically with this setup. Maybe we don't need all the tag names above. 
[Here are some extensive documentation](https://github.com/docker/metadata-action) about the different tag types that can be configured. Lots of reading there.

Reply:
@kappsegla as @JesperBodin mentioned we have now pushed more commits with these requested changes, please confirm if it looks good.